### PR TITLE
feat(manager): create global manager api

### LINKED ISSUE #669: [manager] Create global state manager instance
- it is detached from react rendering cycle
- only serves as an API to get the state data
- does not update the form context at first
  - that will be handled by subscription later on
- add basic API
  - store values
  - change form value
  - registered fields list

### PR DESCRIPTION
**Pull request by Hyperkid123**
closes #669 

- [x] add tests

### CODE REVIEW

File: packages/form-state-manager/src/files/form-manager-context.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ const FormManagerContext = createContext<ManagerContextValue>({
   handleSubmit: noop,
   registerField: noop,
   unRegisterField: noop,
```
Comment (rvsia) [inline packages/form-state-manager/src/files/form-manager-context.ts:15]: It does not belong to this PR... but unregister should be one word `unregisterField`. :smile:

**Comment by Hyperkid123** [comment]
looks like npm is down :question: CI is failing to install dependencies
```
error An unexpected error occurred: "https://registry.yarnpkg.com/@khala/npm-release-monorepo/-/npm-release-monorepo-2.4.1.tgz: Request failed \"404 Not Found\"".
```

File: packages/form-state-manager/src/files/form-manager-context.ts (hunk 1 above)
Comment (Hyperkid123) [reply packages/form-state-manager/src/files/form-manager-context.ts:15]: yup I think this whole state will move to the global manager API. And the context will be eventually updated only when subscribed.

**Comment by Hyperkid123** [comment]
now it fails on: 
```
error An unexpected error occurred: "https://registry.yarnpkg.com/@definitelytyped/utils/-/utils-0.0.34.tgz: Request failed \"404 Not Found\"".
```

I will restart it in a couple of minutes. Locally it installs fine...

**Comment by Hyperkid123** [comment]
@rvsia you can merge it if you want everything is working fine locally and this does not change any other package... IF we want to continue work on this today we will need this PR.
